### DISCUSSION TITLE
Persistent setting for fillcolor on progress bar?

### DISCUSSION THREAD

**Discussion by 7hir7een** [koreader/koreader#8400] [Q&A]
I fully admit I don't know anything about the code language koreader is written in, so this may have an obvious answer.  

On my Kobo Libra 2, _Blitbuffer.COLOR_DARK_GRAY_ is not differentiable enough from _Blitbuffer.COLOR_LIGHT_GRAY_ for me on the thin progress bar.  I did figure out that changing the fillcolor to _Blitbuffer.COLOR_DIM_GRAY_ will provide exactly the contrast I want.  

However, this change is of course lost between each update when changing progresswidget.lua directly.

Is there any code I can put in reader settings or defaults or anywhere else to make this tweak carry through updates?

**Comment by NiLuJe** [comment]
Nope, it's currently hard-coded.

Fairly low hanging branch f someone wants to make it configurable, though, although I'm mildly intrigued by what would make the thin bar different, because IIRC, the only thing using `COLOR_LIGHT_GRAY` is the non linear flow stuff.

(Full disclosure: I haven't looked at the thin bar in a while ^^).

**Comment by NiLuJe** [comment]
Oh, okay, tried it, yeah, might be worth special-casing the color in thin mode, because there's no ticks, so the whole reasoning behind the new lighter color is moot.

Thoughts, @hius07 & the progress bar band? ;).

**Comment by NiLuJe** [comment]
Case in point:

Current:
![Reader_2021-11-01_173211](https://user-images.githubusercontent.com/111974/139707254-927324a6-e4a5-4393-ad3d-aeea1de95e86.png)

Fixed:
![Reader_2021-11-01_173249](https://user-images.githubusercontent.com/111974/139707284-f6bd1e56-7625-4a6e-a901-681af09a31ee.png)

**Comment by hius07** [comment]
I would get rid of the thick/thin distinction and allow user to set all the bar parameters: size, ticks, four colors (bg, border, fill, alt).
It might be a nice radiobuttonwidget to choose one of the eight standard colors.
No more complaints on the contrast.

**Reply by NiLuJe** [reply]
That's not necessarily a bad idea, but fixing the contrast will have to do for now ;p.

**Comment by hius07** [comment]
Since `COLOR_LIGHT_GRAY` is mentioned, is it about non-linear flows?

**Reply by NiLuJe** [reply]
Nope, it's just that thin mode tweaks the bg color, and I hadn't noticed ;p.

Fixed it in https://github.com/koreader/koreader/pull/8393/commits/6464b1df74c73a381ea5962ae37ce145e0d2951e

**Reply by 7hir7een** [reply]
Sorry about the confusion with the LIGHT_GRAY.  As I said, I don't know this code and read into the wrong thing ><.  

But this file is working nicely, thanks for adding!
